### PR TITLE
fix: Convey wallet usability while exporting wallet list to host

### PR DESCRIPTION
**Pull request by kartikCypherock**
Implements https://app.clickup.com/t/9002019994/PRF-5021
Bug fix to correctly convey wallet usability to host. As per Cysync requirements, there is no need to convey the reasoning behind the non-useability.

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-5021 Wallet state communication to host and rejection in case of unusable wallet state](https://app.clickup.com/t/9002019994/PRF-5021)

File: apps/manager_app/export_wallets.c [hunk 1]
```diff
@@ -144,7 +144,17 @@ static void fill_wallet_list(manager_get_wallets_result_response_t *resp) {
     uint8_t wallet_info = get_wallet_info(flash_index);
     list[index].has_passphrase = WALLET_IS_PASSPHRASE_SET(wallet_info);
     list[index].has_pin = WALLET_IS_PIN_SET(wallet_info);
-    list[index].is_valid = (VALID_WALLET == get_wallet_state(flash_index));
+
+    Flash_Wallet *flash_wallet = NULL;
+    get_flash_wallet_by_name(wallet_list[index], &flash_wallet);
+
+    ASSERT(NULL != flash_wallet);
+    if (flash_wallet->is_wallet_locked || VALID_WALLET != flash_wallet->state ||
```
Comment (ujjwal-cyph) [inline apps/manager_app/export_wallets.c:152]: can list the types of wallet states that are non-operational. this is only for completeness and later reference.

example locked, partially synced wallet (wallet without device share), invalid wallet, partially deleted wallet, etc.

**Review by ujjwal-cyph — APPROVED** [review]
LGTM

File: apps/manager_app/export_wallets.c (hunk 1 above)
Comment (kartikCypherock) [reply apps/manager_app/export_wallets.c:157]: added

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**